### DISCUSSION TITLE
Node numbers

### DISCUSSION THREAD

**Discussion by wmboon** [pmgbergen/porepy#757] [Q&A] [answered]
Hi, 

We were used to using the `sd_data["node_number]"` value for the subdomain grids and we see that this has now been removed. Is there an easy way to get the index of a subdomain within a MixedDimensionalGrid?

One cumbersome way we had thought of was, for given sd and mdg:
`node_number = mdg.argsort_grids(mdg.subdomains())[np.array([sd.id for sd in mdg.subdomains()]) == sd.id][0]`

We were hoping that there is a better solution within PorePy.

Your friends in Milano, Alessio and Wietse

**Comment by IvarStefansson** [comment, marked as answer]
I don't know exactly what you mean by "index of sd within mdg", so I'll explain the sorting hoping that helps you.

Domains are sorted first by (descending) dimension and then by (ascending) subdomain id. This might be a definition of the index. Since mdg.subdomains() returns sorted, you also have the index as mdg.subdomains().index(sd). However, for most purposes I think you could probably just use sd.id directly.

Please ask if you need further assistance!

Grüße aus Stuttgart,
Ivar

**Reply by wmboon** [reply]
Thank you, that's exactly what we needed
